### ISSUE TITLE
[BUG]: printf in executors is parsed as errors

### ISSUE THREAD

**Issue by Alcaro** [compiler-explorer/compiler-explorer#4084]
### Describe the bug

What title says

### Steps to reproduce

1.
```c++
int main()
{
    for (int i=1;i<10;i++)
    {
        // this line is fine 🦭
        if (i != 5)
            __builtin_printf("<source>:%d: This line needs more pinnipeds\n", i);
    }
}
```
2. Create an executor. The issue does not reproduce in normal compilers, even if you enable the execute bit.

### Expected behavior

Don't report errors that aren't errors

### Reproduction link

https://godbolt.org/z/67vYYK53c

### Screenshots

![image](https://user-images.githubusercontent.com/5085186/192090037-995db064-b32b-423d-b009-9768f74d1b59.png)

Originally discovered by Ali A. on Discord.

**Comment by RubenRBS** [comment]
I was going to close this as closed in #4080, but that one only disabled the linking from the executor to the editor, not the squiggly generation in the editor itself.
Thanks for the heads-up :)

**Comment by jeremy-rifkin** [comment]
Closing to coalesce with https://github.com/compiler-explorer/compiler-explorer/issues/4179.
